build.mk

```diff
@@ -4,11 +4,11 @@ include $(TOP)/config.mk
 -include $(TOP)/config.override.mk
 
 .PHONY: all
-#all: build check test integration
-all: build test
+#all: build checks test integration
+all: build checks test
```

Comment:
This made me realize that you actually do _not_ include build.mk from the top level.  This means that on the top-level 'checks' is _not_ executed by when you just run 'make' or 'make all'.   Also the build target is currently not recursively triggered.  So i suggest we add also following patch:

```
diff --git a/Makefile b/Makefile
index 809dd37..317636e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,9 @@
+TOP = .
+include $(TOP)/build.mk
+
 SUB_DIRS = utils/fabric-ccenv-sgx ercc ecc_enclave ecc tlcc_enclave tlcc

-all clean test:
+all build clean test:
        $(foreach DIR, $(SUB_DIRS), $(MAKE) -C $(DIR) $@;)

 checks: linter

```

Reply:
thanks! added